core/modules/comment/comment.module

```diff
@@ -724,7 +724,7 @@ function comment_node_page_additions(Node $node) {
   // Only attempt to render comments if the node has visible comments.
   // Unpublished comments are not included in $node->comment_count, so show
   // comments unconditionally if the user is an administrator.
-  if (($node->comment_count && user_access('access comments')) || user_access('administer comments')) {
+  if ((!empty($node->comment_count) && $node->comment_count && user_access('access comments')) || user_access('administer comments')) {
```

Comment:
```
!empty($node->comment_count) && $node->comment_count
```

The second conditional here is redundant with the first one. It can be simplified down to just:

```
!empty($node->comment_count)
```